File: backend/src/main/kotlin/at/wrk/tafel/admin/backend/common/api/PagedResponse.kt
package at.wrk.tafel.admin.backend.common.api

import at.wrk.tafel.admin.backend.common.ExcludeFromTestCoverage

@ExcludeFromTestCoverage
data class PagedResponse<T>(
    val items: List<T>,
    val totalCount: Long,
    val currentPage: Int,
    val totalPages: Int,
    val pageSize: Int,
)

File: frontend/src/main/webapp/src/app/common/api/paged-response.ts
export interface PagedResponse<T> {
  items: T[];
  totalCount: number;
  currentPage: number;
  totalPages: number;
  pageSize: number;
}

export const PAGE_SIZE_OPTIONS = [5, 10, 25, 50, 100];
export const DEFAULT_PAGE_SIZE = 10;
